cmd/plugin/delete_old_txt.go

```diff
@@ -0,0 +1,203 @@
+			fmt.Printf("unexpected nmber of secrets: %d; expected 1 default secret\n", len(secretList.Items))
+			return nil
+		}
+		secret = &secretList.Items[0]
+
+	} else {
+		err = k8sClient.Get(ctx, client.ObjectKey{Name: args[0], Namespace: deleteOldTXTCMDFlags.ns}, secret)
+		if err != nil {
+			fmt.Printf("failed to get secret: %v\n", err)
+			return nil
+		}
+	}
+	// factory to get a list of all endpoints
+	dynClient, err := dynamic.NewForConfig(config)
+	if err != nil {
+		fmt.Printf("failed to create dynamic client: %v\n", err)
+		return nil
+	}
+
+	providerFactory, err := provider.NewFactory(k8sClient, dynClient, provider.RegisteredDefaultProviders(), endpoint.NewAuthoritativeDNSRecordProvider)
+	if err != nil {
+		fmt.Printf("failed to create provider factory: %v\n", err)
+		return nil
+	}
+
+	// empty config to list all records
+	endpointProvider, err := providerFactory.ProviderForSecret(ctx, secret, provider.Config{})
+	if err != nil {
+		fmt.Printf("failed to get provider: %v\n", err)
+		return nil
+	}
+
+	endpoints, err := endpointProvider.Records(ctx)
+	if err != nil {
+		fmt.Printf("failed to get endpoints: %v\n", err)
+		return nil
+	}
+
+	// only old TXTs created by us should be left
+	endpoints = slice.Filter(endpoints, func(e *externaldns.Endpoint) bool {
+		if e != nil &&
+			e.RecordType == externaldns.RecordTypeTXT &&
+			strings.HasSuffix(e.DNSName, deleteOldTXTCMDFlags.domain) {
+
+			// we will always have at least one target
+			if len(e.Targets) == 0 {
+				return false
+			}
+
+			// two layers of checks to make sure we don't delete anything needed
+
+			// owner and version in targets
+			owner, epVersion, epLabels, err := registry.NewLabelsFromString(e.Targets[0], []byte{})
+			if err != nil {
+				fmt.Printf("failed to extract owner labels: %v\n", err)
+				return false
+			}
+
+			// happens when there was only one owner
+			if owner == "" {
+				owner = epLabels[externaldns.OwnerLabelKey]
+			}
+
+			// it is old only if it has owner and doesn't have version
+			if owner != "" && epVersion == "" {
+				// this should be redundant,
+				// but it might happen that they have TXTs from external dns that are not from our operator
+				// make sure the name of endpoint matches our old pattern
+				mapper := registry.NewExternalDNSAffixNameMapper("kuadrant-", "", "wildcard")
+				epName, recordType := mapper.ToEndpointName(e.DNSName, epVersion)
+
+				// if they aren't empty - it was our record
+				return epName != "" && recordType != ""
+			}
+			return false
+		}
+		return false
+	})
+
+	fmt.Printf("filtered endpoints len %d\n", len(endpoints))
+	for _, ep := range endpoints {
+		fmt.Printf("ep: %s\n", ep.DNSName)
+	}
+
+	// display what is about to be deleted
+	fmt.Printf("TXT records (%d) to be deleted:\n", len(endpoints))
+	for _, txtRecord := range endpoints {
+		fmt.Printf("%s\t IN\t %s\t %s\n", txtRecord.DNSName, txtRecord.RecordType, txtRecord.Targets)
+	}
+	fmt.Printf("Do you want to proceed? [Y/N]\n")
```

Comment:
Its a bit odd to put the user input on a new line.
```suggestion
	fmt.Printf("Do you want to proceed? [Y/N]: ")
```

Reply:
Had a though here. Can we add a confirm flag to over ride this prompt check. This user required input means this cli command can never be used in a ci/cd pipeline.

Reply:
hmm, I'd be extremely concerned if anyone would try to have a tool that deletes records somewhere in a pipeline. 
The reason I go for the input is to make sure the user is aware of what is happening and not just fat-fingered their way with autocomplete here and nuked their whole TXT registry of something. 

That said I see no harm in having a flag (like [dnf](https://dnf.readthedocs.io/en/latest/command_ref.html) has, for example)